packages/delegation-process/test/revokeDelegation.test.ts

```diff
@@ -0,0 +1,266 @@
+
+    expect(lastDelegationEvent.version).toBe("1");
+    expect(delegationFromLastEvent).toMatchObject(expectedDelegation);
+
+    vi.useRealTimers();
+  });
+
+  it("should throw an delegationNotFound if Delegation not exists", async () => {
+    const delegatorId = generateId<TenantId>();
+    const authData = getRandomAuthData(delegatorId);
+    const delegationId = generateId<DelegationId>();
+    await expect(
+      delegationProducerService.revokeDelegation(delegationId, {
+        authData,
+        logger: genericLogger,
+        correlationId: randomUUID(),
+        serviceName: "DelegationServiceTest",
+      })
+    ).rejects.toThrow(delegationNotFound(delegationId));
+  });
+
+  it("should throw an delegatorNotAllowToRevoke if Requester Id and DelegatorId are differents", async () => {
+    const currentExecutionTime = new Date();
+    vi.useFakeTimers();
+    vi.setSystemTime(currentExecutionTime);
+
+    const delegatorId = generateId<TenantId>();
+    const delegateId = generateId<TenantId>();
+    const authData = getRandomAuthData(delegatorId);
+    const delegationId = generateId<DelegationId>();
+
+    const delegationCreationDate = new Date();
+    delegationCreationDate.setMonth(currentExecutionTime.getMonth() - 2);
+
+    const delegationApprovalDate = new Date();
+    delegationApprovalDate.setMonth(currentExecutionTime.getMonth() - 1);
+
+    const existentDelegation = {
+      ...getMockDelegationProducer({
+        id: delegationId,
+        delegateId,
+      }),
+      approvedAt: delegationApprovalDate,
+      submittedAt: delegationCreationDate,
+      stamps: {
+        submission: {
+          who: delegatorId,
+          when: delegationCreationDate,
+        },
+        approval: {
+          who: delegateId,
+          when: delegationApprovalDate,
+        },
+      },
+    };
+
+    await addOneDelegation(existentDelegation);
+
+    await expect(
+      delegationProducerService.revokeDelegation(delegationId, {
+        authData,
+        logger: genericLogger,
+        correlationId: randomUUID(),
+        serviceName: "DelegationServiceTest",
+      })
+    ).rejects.toThrow(delegatorNotAllowToRevoke(existentDelegation));
+    vi.useRealTimers();
+  });
+
+  it("should throw an delegatorNotAllowToRevoke if delegation doesn't have revocable one of revocable states [Rejected,Revoked]", async () => {
+    const currentExecutionTime = new Date();
+    vi.useFakeTimers();
+    vi.setSystemTime(currentExecutionTime);
+
+    const delegatorId = generateId<TenantId>();
+    const delegateId = generateId<TenantId>();
+    const authData = getRandomAuthData(delegatorId);
+
+    const delegationCreationDate = new Date();
+    delegationCreationDate.setMonth(currentExecutionTime.getMonth() - 2);
+
+    const delegationActivationDate = new Date();
+    delegationActivationDate.setMonth(currentExecutionTime.getMonth() - 1);
+
+    const notRevocableDelegationState = getNotRevocableStateSeeds();
```

Comment:
instead of using `getNotRevocableStateSeeds` how about we use the it.each construct and build the seed based on the given state? In this way we can always test both states instead of randoming between the two

Reply:
done in this [7605b94](https://github.com/pagopa/interop-be-monorepo/pull/1100/commits/7605b9460d09f6fae754a893a4818a0cb12ccfac)